rfd/0036/user-stories/jupiter-example-com-jenkins.md

```diff
@@ -0,0 +1,77 @@
+
+
+### Jenkins production job
+
+Despite the automation for continuous build and deploy of the staging projects, the deploys to the live site require manual promotion. After reviewing the relevant staging site and any logs generated by Jenkins, the operators merge a PR to master, triggering a new build of master that is tagged with the date and time of the build (the builds collect resources beyond the scope of the git repo, so the git SHA is not a consistent identifier). Jenkins stages that build in a project named <repo-name>-branch-<branch name> (using the same logic as the development job above). After testing and previewing that final staging build, the operators can choose to manually deploy that build to the live site. This manual deployment is done via a separate production job.
+
+This Jenkins job is not configured to receive GitHub webhook pushes. Instead, it is a parameterized job which deploys the parameterized tag (with a default like “production”). The end-user tags the release in git with a version and the tag.
+
+```
+git tag v1.2.3
+git tag production
+git push --tags
+```
+
+The operator uses the Jenkins UI (or CLI) to start the deployment job with the particular tag needed (with the default being part of the Jenkins job definition). The job tags the correct version as “production” and renders the production manifest with the new version and pushes it to Mariposa for deployment. Note that Mariposa isn’t watching registry server; you need to explicitly update images in IMGAPI to get new images, so updating the Docker image tag in the registry is not enough to trigger a deployment.
+
+> Note: We should surface the content hashes in the UI so that it’s obvious if a deploy with different image ID but the same Docker tag has occurred.
+
+### Secrets
+
+There are a small number of sensitive details that need to be carefully handled, including the SSL cert and key for Nginx, possibly TLS client and server certs for Consul, and a license key for the CMS. The application operators expect Mariposa to offer features to simplify that process.
+
+Secrets can be injected by Mariposa into containers via instance metadata. These can be accessed via ContainerPilot preStart or the application.
+
+In hardened environments the instance metadata could be a one-time token for something like Vault. The scheduler would need to coordinate with w/ the external secrets management server though (maybe for a post-MVP).
+
+Jenkins would generate secrets as part of the job for the dev environment, add those to the manifest’s metadata (this isn’t in the manifest file itself!). This allows the dev credentials be ephemeral for each deployment.
+
+For production metadata, the operators can add those secrets to Mariposa out-of-band from the CI server. This way Jenkins never gets the production secrets. It only has the manifest not the metadata for production. (Although in practice in has the keys to the universe until RBAC v2.)
```

Comment:
> Although in practice in has the keys to the universe until RBAC v2

We need to get more particular about what permissions are required to support this process